### PR TITLE
Adds chemglands

### PR DESCRIPTION
**Pull request by FuryMcFlurry**
- rscadd: "Added experimental organ, strange organs of unknown origins, that contains 2 chemicals and pumps it into your system when you have the organ implanted (huge thanks to ArcLumin/McBawbaggins/TDcoolguy300 for helping me understand code a little more. requires CMO access to unlock the crate."
- imageadd: "Sprites done by me."
- rscadd: "Added experimental organ crates for Cargo to order."
- rscadd: "Experimental organs can also be found in maint... sometimes."

screenshot_20

I dont know how to git hub pls no bully
I have tested the glands and they work.

### CODE REVIEW

**Comment by KayeArray** [comment]
Sounds like fun

**Comment by ArcLumin** [comment]
If you want, add them to maint loot spawners as a small chance

**Comment by FuryMcFlurry** [comment]
I first have to learn how to add lines insted of uploading by file as it straight up replaces the file.

**Comment by ArcLumin** [comment]
Just copy and paste the text

**Comment by ArcLumin** [comment]
Code is good

**Comment by blobbernaut** [comment]
The chance to add some chems is only 5%, should increase that chance.

**Comment by FuryMcFlurry** [comment]
@blobbernaut  trust me dude, having it higher makes bad chemicals fuck you over real fast, death in about 5 mins.

**Comment by doobiedoo23** [comment]
Sure +1

File: code/modules/surgery/organs/organ_internal.dm [hunk 1]
```diff
@@ -1,236 +1,265 @@
...
+	name = "[pick("spiky","twisted","odd","horrific","bloated","tumor","tumorous","deformed","cancerous","smooth","convulsing","fleshy","organic","alien","lumpy","warped")][pick("-like"," looking","")] [pick("organ","tissue","gland")]"
+	var/list/chemgland_chems = list("rezadone", "histamine", "hippiesdelight", "ethanol", "mushroomhallucinogen", "heroin", "bath_salts", "methamphetamine", "crank", "nicotine", "space_drugs", "aranesp", "carrotjuice", "krokodil", "fartium", "laughter", "colorful_reagent", "serotrotium", "mulligan", "holywater", "histamine", "toxinsspecial", "hell_ramen", "venom", "cyanide", "sulfonal", "lipolicide", "heparin", "teslium", "rotatium", "bleach", "unholywater", "mercury", "frostoil", "mine_salve", "calomel", "sleeptoxin", "mutagen", "lexorin", "mindbreaker", "spore", "spore_burning", "tirizene", "synaptizine", "icecoffee", "coffee", "hot_coco", "leporazine", "oxandrolone", "charcoal", "potass_iodide", "sal_acid", "salbutamol", "perfluorodecalin", "ephedrine", "oculine", "atropine", "mannitol", "antihol", "insulin", "kelotane", "dexalin", "mannitol", "bromelain", "cafe_latte")
```
Comment (ghost) [inline code/modules/surgery/organs/organ_internal.dm]: mannitol appears twice

**Comment by ghost** [comment]
Don't like how there are some pretty potent chemicals both ways.

I could totally see this as a job exclusive CMO item that gives crazy chems though.

File: code/modules/surgery/organs/organ_internal.dm (hunk 1 above)
Comment (ArcLumin) [reply code/modules/surgery/organs/organ_internal.dm]: double the chance to not be retarded

**Comment by ArcLumin** [comment]
considering it adds it to your body, you don't know what it is until you get it in you. So you don't know if it's a super good healing item or deadly poison. Or both

**Comment by KayeArray** [comment]
It's balanced via randomness, I wouldn't trust that shit since some of those chems are truly horrible

**Comment by ArcLumin** [comment]
I mean, I'd be hella sus of poppin that in me, but I might do it for a fast check

**Comment by ghost** [comment]
I think it's a bad idea to add some seriously gamebreaking chems that could be easily accessible.
You could bypass the risk factor by having a _willing_ subject (i.e. a monkey human.)

**Comment by ArcLumin** [comment]
still think it'd be great if it chose two random chems of every single chem in the game sans admindrozine

**Comment by FuryMcFlurry** [comment]
reminder that the prob is super low anyway.

**Comment by ghost** [comment]
I think there's a cool idea on this, I'm just skeptic on the current implementation.

**Comment by FuryMcFlurry** [comment]
what  if i were to turn the price up
like crazy.

**Comment by blobbernaut** [comment]
Even adminordrazine would be fine

**Comment by ghost** [comment]
> "Even adminordrazine would be fine"

I'm sorry what

**Comment by KayeArray** [comment]
That's basically the admin heal command in reagent form.

nope

**Comment by blobbernaut** [comment]
5% chance to get a single unit of it per tic, with the chance to even get it across all the chems in the game and thus super low?

Wouldn't be that op.

**Comment by blobbernaut** [comment]
It can't heal being dead.

**Comment by ArcLumin** [comment]
but it can. It also heals lost limbs if I recall.

**Comment by blobbernaut** [comment]
It REPAIRS corpses, yes. But it doesn't set your stat to being back alive IIRC.

**Comment by blobbernaut** [comment]
/datum/reagent/medicine/adminordrazine/on_mob_life(mob/living/carbon/M)
	M.reagents.remove_all_type(/datum/reagent/toxin, 5*REM, 0, 1)
	M.setCloneLoss(0)
	M.setOxyLoss(0)
	M.radiation = 0
	M.heal_organ_damage(5,5,2)
	M.adjustToxLoss(-5)
	M.hallucination = 0
	M.setBrainLoss(0)
	M.disabilities = 0
	M.eye_blurry = 0
	M.eye_blind = 0
	M.SetWeakened(0)
	M.SetStunned(0)
	M.SetParalysis(0)
	M.silent = 0
	M.dizziness = 0
	M.drowsyness = 0
	M.stuttering = 0
	M.slurring = 0
	M.confused = 0
	M.sleeping = 0
	M.jitteriness = 0
	for(var/datum/disease/D in M.viruses)
		if(D.severity == NONTHREAT)
			continue
		D.spread_text = "Remissive"
		D.stage--
		if(D.stage < 1)
			D.cure()
	..()
	return

**Comment by ArcLumin** [comment]
Sets oxyloss to 0, 5 of all damages healed every tick, immune to all stuns. Its definately spaceproof, even completely naked

**Comment by blobbernaut** [comment]
I think brute damage might be very slightly faster than that.

**Comment by ArcLumin** [comment]
The pressure deals brute, the temp deals burn. It's not faster than that.

**Comment by blobbernaut** [comment]
Ah. I found it.

If you're breathing in cold, you take damage to your head. If you're just cold, you take damage to your chest. In space it's both, so you take the level 3 cold damage twice, which adds to 6 per tic.

So technically you're taking 1 burn damage per tic.

**Comment by ArcLumin** [comment]
So spaceworthy with internals

**Comment by FuryMcFlurry** [comment]
Changes the price to 55 and added cmo access onto the crate.

**Comment by ArcLumin** [comment]
You gonna keep it a whitelist or make it a blacklist for all chems but admindrozine?

**Comment by FuryMcFlurry** [comment]
gonna keep it as it is. dont think tricord and other all round healing chemicals works.
@McBawbaggings Also as an argument how strong this could be. take a look at the hivelord core, that shit is almost guaranteed all damage heals at regular ticks, actually helps people out of crit. so this organ cant be that bad by comparison.

**Comment by blobbernaut** [comment]
Honestly I don't know why you care about blacklisting adminordrazine. It's just a souped up regeneration chemical with stun immunity. When chemicals like xenomicrobes exist for example.

**Comment by FuryMcFlurry** [comment]
if this goes live you can make a pull request to add that chemical. and lets leave it at that.

**Comment by blobbernaut** [comment]
i don't think you get the context of what i was talking about fury

somebody else mentioned just making it take from all chemicals in the game randomly, except adminordrazine and i was just pointing out that it wasn't really that op

**Comment by FuryMcFlurry** [comment]
oh alright.

**Comment by Pyko1** [comment]
Oh god, this in a monkey human + Oddy chem duper.

**Comment by ghost** [comment]
I'm willing to trial this, even if it's a bit broken it's not going to be super OP.
(Hopefully)

**Comment by Lawligagger** [comment]
Willing to give a go as well

**Comment by FuryMcFlurry** [comment]
if doobie is a headmin that be 3 people willing to give it a try.

**Comment by ghost** [comment]
Doobie is a GM

**Comment by FuryMcFlurry** [comment]
well then 1 more headmin and the pact is done.

**Comment by SplergeBucket** [comment]
sure

**Comment by FuryMcFlurry** [comment]
that would be 3 people

File: code/datums/supplypacks.dm [hunk 2]
```diff
@@ -1,1353 +1,1366 @@
-//SUPPLY PACKS
```
Comment (Gandalf2k15) [inline code/datums/supplypacks.dm:1]: This makes me salty. OCD.

**Comment by Gandalf2k15** [comment]
Das code ist gut. Apart from github being a twat.

**Comment by FuryMcFlurry** [comment]
Im sorry but I learned from it, dont upload fiels, just add them manually

**Comment by Gandalf2k15** [comment]
All requirements met. 2/3 headmin approval. Code passed. Merging.

**Pull request merged**